### DISCUSSION TITLE
反向代理：bridge在国内，portal在国外，哪种协议可行？

### DISCUSSION THREAD

**Discussion by upsangel** [XTLS/Xray-core#4057] [Q&A] [answered]
看了 https://youtu.be/7VVMTX8iXbg 想利用大陆家里的宽带（没有公网IP）部署一个逆向回国用的反向代理，如果portal是在国外，那么流量应该也要符合 GFW 规范吧？ 这样的话，用shadowsocks 做反代协议的特征会不会太明显了？ 但是如果用 reality 的话，由于存在“借证书” 动作，好像不太符合反代的工作原理？还是说 reality 也是最优解？ 如下图：

 家里宽带 -> 反向代理 reallity ->  国外VPS

如果不行的话，就要在国内多买一个VPS 变成： 
家里宽带 -> 反向代理 shadowsocks -> 国内VPN -> 正向代理 reallity -> 国外VPS 
这也太折腾了吧。。。？

求解惑

**Comment by onionknight114** [comment, marked as answer]
针对目前tls over tls墙并没有大规模封禁的情况下，我个人的回国反向代理的方法是用的vless+ws+tls+CDN的方法解决的。国外客户端到中转服务器的protal采用shadowsocks获得最优性能，中转服务器向国内的bridge用的vless+ws+tls+CF CDN，国内家庭落地使用本地部署3x-ui然后上行到CDN（甚至还可以优选一下ip），可以完美解决无公网IP的苦恼，实际测试下来几乎可以跑满国内家宽小水管上行

**Reply by upsangel** [reply]
謝謝你的回覆。我找朋友試了下直接 shadowsocks 反代果然幾分鐘就被嚴重限流了。 我研究研究vless+ws+tls 然後再怎麼套 CDN，唯一不好處就是應該要域名了？

**Reply by upsangel** [reply]
我再研究了一下，發現 vmess + ws + tls 好像比較容易在 3xui 面板中配置。 不過vmess不能套vision，現在看看vless+tls+vision應該怎麼配。

**Reply by onionknight114** [reply]
根据官方文档，vision流控并不支持ws和VMess协议，vless+tls+vision是可行的搭配；不过使用vision/reality需要国内家宽落地机--墙外中转的上行网络线路质量优良，且此组合不能搭配任何CDN。如果使用vless+ws（或者XHTTP/SplitHTTP）+tls+CDN的话会出现tls over tls问题(只能算是一种隐患，并无大规模封禁报告），但是优选CDN ip的情况下对中转服务器的要求很低，实现成本低廉，目前我给海外朋友使用的就是这个方案，不知道是否对你有所帮助

**Reply by upsangel** [reply]
> 优选CDN ip的情况下对中转服务器的要求很低，实现成本低廉

謝謝你的回覆。原來如此，我今天的結果也覺得優選IP應該很重要。

嘗試了兩種 
vless + tls （自己域名的證書） + vision ： 國內機用V2RAY直連可以，但是放在3xui中設置爲反代就不行（127.0.0.1 端口監聽失敗，已經bind了不能重複bind之類的報錯），沒有深究到底是什麼原因

vmess + ws + tls （自己域名的證書） ： 國內機用V2RAY直連可以（上傳有30Mbps），放在 3xui 中反代連線也一直可用，但是從中轉機反代回去的速度慢而且不穩定得感人（下載在3-20Mbps間跳動，多數時間是在單位數）。基本上只能停留在 文字 圖片， APP和視頻大機率讀不了或者體驗極差。試了普通套CF，沒改善，可能要優選IP才有效。

**Reply by l3onttt** [reply]
> 针对目前tls over tls墙并没有大规模封禁的情况下，我个人的回国反向代理的方法是用的vless+ws+tls+CDN的方法解决的。国外客户端到中转服务器的protal采用shadowsocks获得最优性能，中转服务器向国内的bridge用的vless+ws+tls+CF CDN，国内家庭落地使用本地部署3x-ui然后上行到CDN（甚至还可以优选一下ip），可以完美解决无公网IP的苦恼，实际测试下来几乎可以跑满国内家宽小水管上行

“国外客户端到中转服务器的protal采用shadowsocks获得最优性能”，为啥要用ss，中转和客户端都在墙外，为啥不直接用socks代理啊？ss有啥优势吗？

**Reply by onionknight114** [reply]
socks代理传输本身没有任何加密，出于基本的数据安全性考虑，用自带加密且性能影响小、部署简便的ss协议更合适
